Android/src/org/droidplanner/android/maps/providers/baidu_map/BaiduMapOfflineMapPreference.java

```diff
@@ -0,0 +1,357 @@
+import android.util.AttributeSet;
+import android.view.View;
+import android.view.View.OnClickListener;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.BaseAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.LinearLayout;
+import android.widget.ListAdapter;
+import android.widget.ListView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.baidu.mapapi.map.offline.MKOLSearchRecord;
+import com.baidu.mapapi.map.offline.MKOLUpdateElement;
+import com.baidu.mapapi.map.offline.MKOfflineMap;
+import com.baidu.mapapi.map.offline.MKOfflineMapListener;
+
+import java.util.ArrayList;
+
+import org.droidplanner.android.R;
+
+public class BaiduMapOfflineMapPreference extends DialogPreference implements MKOfflineMapListener {
+    private MKOfflineMap mOffline = null;
+    private View mBindDialogView = null;
+    private TextView mCidView = null;
+    private TextView mStateView =null;
+    private EditText mCityNameView = null;
+    private LocalMapAdapter mLAdapter = null;
+    private LinearLayout mCityListLayout = null;
+    private LinearLayout mLocalMapLayout = null;
+    private ArrayList<MKOLUpdateElement> mLocalMapList = null; // downloaded offline map list
+    private static final int DATA_SIZE_ONE_KB = 1024 * 1024;
+    private static final int DATA_SIZE_UNIT_KB = 1024;
+    private static final double DATA_SIZE_UNIT_MB = 1024 * 1024.0;
+
+    public BaiduMapOfflineMapPreference(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        setDialogLayoutResource(R.layout.fragment_baidumap_offline_map_preference);
+    }
+
+    private void initView(View view) {
+
+        mCidView = (TextView) view.findViewById(org.droidplanner.android.R.id.cityid);
+        mCityNameView = (EditText) view.findViewById(org.droidplanner.android.R.id.city);
+        mStateView = (TextView) view.findViewById(org.droidplanner.android.R.id.state);
+
+        ListView hotCityList = (ListView) view.findViewById(org.droidplanner.android.R.id.hotcitylist);
+        ArrayList<String> hotCities = new ArrayList<String>();
+        // get hot city list
+        ArrayList<MKOLSearchRecord> records1 = mOffline.getHotCityList();
+        if (records1 != null) {
+            for (MKOLSearchRecord r : records1) {
+                hotCities.add(r.cityName + "(" + r.cityID + ")" + "   --" + this.formatDataSize(r.size));
+            }
+        }
+        ListAdapter hAdapter = (ListAdapter) new ArrayAdapter<String>(this.getContext(),
+                android.R.layout.simple_list_item_1, hotCities);
+        hotCityList.setAdapter(hAdapter);
+
+        ListView allCityList = (ListView) view.findViewById(R.id.allcitylist);
+        // get all offline city map
+        ArrayList<String> allCities = new ArrayList<String>();
+        ArrayList<MKOLSearchRecord> records2 = mOffline.getOfflineCityList();
+        if (records1 != null) {
```

Comment:
@droneboost you're checking if `records1` is not `null` instead of `records2`.
Asides from the fix, can you also rename these two variables to improve legibility.